lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -106,15 +106,27 @@ export class PublicClientApplication {
      * containing data from the server (returned with a null or non-blocking error).
      */
     async handleRedirectCallback(authCallback: AuthCallback): Promise<void> {
-        if(!authCallback) {
+        if (!authCallback) {
             throw BrowserConfigurationAuthError.createInvalidCallbackObjectError(authCallback);
         }
 
         this.authCallback = authCallback;
         const { location: { hash } } = window;
-        if (UrlString.hashContainsKnownProperties(hash)) {
-            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback);
-            interactionHandler.handleCodeResponse(hash);
+        const cachedHash = this.browserStorage.getItem(TemporaryCacheKeys.URL_HASH);
+        try {
+            const interactionHandler = new RedirectHandler(this.authModule, this.browserStorage, this.authCallback, this.config.auth.navigateToLoginRequestUrl);
+            let tokenResponse: TokenResponse;
+            if (UrlString.hashContainsKnownProperties(hash)) {
+                tokenResponse = await interactionHandler.handleCodeResponse(hash);
+            } else if (UrlString.hashContainsKnownProperties(cachedHash)) {
+                tokenResponse = await interactionHandler.handleCodeResponse(cachedHash);
+            }
```

Comment:
```js
const responseHash = UrlString.hashContainsKnownProperties(hash) ? hash : cachedHash;
const tokenResponse = await interactionHandler.handleCodeResponse(responseHash);
```

Reply:
+1

Reply:
done.

Reply:
This doesn't work, since there are two predicates being checked here. We don't set the token response if there is a hash in the window or cached in storage.

Reply:
```js
const responseHash = UrlString.hashContainsKnownProperties(hash) ? hash : cachedHash;

if (responseHash) {
  const tokenResponse = await interactionHandler.handleCodeResponse(responseHash);

  if (tokenResponse) {
     this.authCallback(null, tokenResponse);
  }
}
```

Reply:
I don't really see the value add of this but I will go ahead and make the change.

Reply:
done.